src/algebra/associated.lean

```diff
@@ -224,6 +228,10 @@ iff.intro
   (assume h, let ⟨c, h⟩ := h.symm in h ▸ ⟨c, (one_mul _).symm⟩)
   (assume ⟨c, h⟩, associated.symm ⟨c, by simp [h]⟩)
 
+lemma is_unit_of_associated_is_unit [monoid α] {p q : α}
+  (h : associated p q) (hp : is_unit p) : is_unit q :=
```

Comment:
We already have `associated.is_unit` and `associated.is_unit_iff`.

Reply:
Oh I missed that, my bad!